### PR TITLE
remove SMP_PRESENT and replace with BUILD_THREADED

### LINKED ISSUE #4544: BUILD_THREADED or SMP_PRESENT
It seems that there is a confusion in the usage of these two variables and 
I think that we should remove one of them.   A grep of BUILD_THREADED in the CESM source tree reveals that it is used in a number of components and libraries but does not
seem to be used in cime, so I propose removing this variable and replacing it with SMP_PRESENT throughout.

### PR DESCRIPTION
**Pull request by jedwards4b**
SMP_PRESENT and BUILD_THREADED were both used in the same way and having two variables to do the same thing is confusing at best.  This replaces all instances of SMP_PRESENT with BUILD_THREADED.   Note that this may require changes in E3SM outside of CIME.

Test suite: scripts_regression_tests.py
Test baseline:
Test namelist changes:
Test status: bit for bit
Fixes #4544 
User interface changes?: xml variable SMP_PRESENT is removed. 

Update gh-pages html (Y/N)?:

### CODE REVIEW

**Comment by jedwards4b** [comment]
system-testing expected to fail

**Review by jgfouca — APPROVED** [review]
Thanks, I think this is a good change.

**Comment by amametjanov** [comment]
@jgfouca, will there be a follow-up PR to fix things in E3SM?
Just wanted to check that `BUILD_THREADED` was set F->T->F with NTHREADS changing 1->2->1: noted in  https://github.com/ESMCI/cime/pull/2466#issuecomment-380939087 .
```
azamat@perlmutter:login09:~/saul/E3SM
> cd cime && git branch && cd -
  master
* replace_smp_present
azamat@perlmutter:login09:~/saul/E3SM
> ./cime/scripts/create_test SMS.f19_g16.X --no-build -t 20231218-chk-cime
Using project from .cesm_proj: e3sm
create_test will do up to 1 tasks simultaneously
create_test will use up to 320 cores simultaneously
Creating test directory /pscratch/sd/a/azamat/e3sm_scratch/pm-cpu/SMS.f19_g16.X.pm-cpu_intel.20231218-chk-cime
RUNNING TESTS:
  SMS.f19_g16.X.pm-cpu_intel
Starting CREATE_NEWCASE for test SMS.f19_g16.X.pm-cpu_intel with 1 procs
Finished CREATE_NEWCASE for test SMS.f19_g16.X.pm-cpu_intel in 5.232196 seconds (PASS)
Starting XML for test SMS.f19_g16.X.pm-cpu_intel with 1 procs
Finished XML for test SMS.f19_g16.X.pm-cpu_intel in 0.363084 seconds (PASS)
Starting SETUP for test SMS.f19_g16.X.pm-cpu_intel with 1 procs
Finished SETUP for test SMS.f19_g16.X.pm-cpu_intel in 1.173132 seconds (FAIL). [COMPLETED 1 of 1]
    Case dir: /pscratch/sd/a/azamat/e3sm_scratch/pm-cpu/SMS.f19_g16.X.pm-cpu_intel.20231218-chk-cime
    Errors were:
        ERROR: No variable BUILD_THREADED found in case
```

**Review by amametjanov — APPROVED** [review]
Checked in E3SM with `SMS.f19_g16.X.pm-cpu_intel` and
```
azamat@perlmutter:login09:~/saul/E3SM
> git diff driver-mct/cime_config/config_component.xml
diff --git a/driver-mct/cime_config/config_component.xml b/driver-mct/cime_config/config_component.xml
index 7c8e83fa70..ea1de9ee8d 100644
--- a/driver-mct/cime_config/config_component.xml
+++ b/driver-mct/cime_config/config_component.xml
@@ -717,7 +717,7 @@
       If TRUE, the component libraries are always built with OpenMP capability.</desc>
   </entry>

-  <entry id="SMP_PRESENT">
+  <entry id="BUILD_THREADED">
     <type>logical</type>
     <valid_values>TRUE,FALSE</valid_values>
     <default_value>FALSE</default_value>
```
- Changing `NTHRDS_ATM` 1->2->1 or `FORCE_BUILD_SMP` F->T->F switches `BUILD_THREADED` F->T->F.
- Need to do the rename during the merge to E3SM.

**Comment by jedwards4b** [comment]
@amametjanov thank you for confirming.

**Comment by jedwards4b** [comment]
@jasonb5 do i need to fix something here or is the ball in your court?

**Comment by jasonb5** [comment]
@jedwards4b It's in mine, I'll restart the tests shortly.

**Comment by jasonb5** [comment]
@jedwards4b [ccs_config](https://github.com/ESCOMP/CESM/blob/c69ce9f93f1b38f1f53a4e73371acf4ccb39576f/Externals.cfg#L1-L6) needs to be updated in the CESM repo.

**Comment by jedwards4b** [comment]
@jasonb5 The file Externals.cfg that is used by docker needs to be updated.

**Comment by jasonb5** [comment]
@jedwards4b I updated the container to use `Externals.cfg` from the CIME repo for both `sys` and `unit` tests. The remaining errors look related to `CDEPS`.
```
SMS.f19_g16_rx1.A.docker_gnu.fake_testing_only_20231220_171621/bld/gnu/openmpi/nodebug/nothreads/nuopc/CDEPS', error=/__w/cime/cime/cime/components/cdeps/streams/dshr_stream_mod.F90:1714:28:

 1714 |     use shr_file_mod, only : shr_file_get_real_path
      |                            1
Error: Symbol 'shr_file_get_real_path' referenced at (1) not found in module 'shr_file_mod'
make[2]: *** [streams/CMakeFiles/streams.dir/build.make:101: streams/CMakeFiles/streams.dir/dshr_stream_mod.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:375: streams/CMakeFiles/streams.dir/all] Error 2
make: *** [Makefile:124: all] Error 2
```

**Comment by jedwards4b** [comment]
I've been trying to figure that out.   
```
 git blame shr_file_mod.F90 | grep shr_file_get_ 
961411df src/shr_file_mod.F90  (Jim Edwards    2023-10-27 13:23:12 -0600   64)   public :: shr_file_get_real_path ! Get a fully resolved path
961411df src/shr_file_mod.F90  (Jim Edwards    2023-10-27 13:23:12 -0600 1010)   subroutine shr_file_get_real_path(path, resolved_path)
961411df src/shr_file_mod.F90  (Jim Edwards    2023-10-27 13:23:12 -0600 1043)   end subroutine shr_file_get_real_path
```

**Comment by jedwards4b** [comment]
Ah - the share should be share1.0.18 - we had it set to 17

**Comment by jedwards4b** [comment]
The e3sm test is expected to fail because the mct driver config_compset.xml file needs to be changed to use BUILD_THREADED instead of SMP_PRESENT.

File: CIME/case/case_clone.py [hunk 1]
```diff
@@ -219,8 +219,6 @@ def create_clone(
             )
         )
 
-        newcase.case_setup()
```
Comment (rljacob) [inline CIME/case/case_clone.py:222]: Did this have anything to do with the SMP_PRESENT to BUILD_THREADED change?

Comment (jedwards4b) [reply CIME/case/case_clone.py:222]: I think it was something I caught as being out of place during the development of that change, so it's not directly related to that but it was probably included in that PR.   If you want to put it back and then open an issue to fix the problem by fixing the tests then I guess we can do it that way...

Comment (rljacob) [reply CIME/case/case_clone.py:222]: Yes I'd like to do it that way.

Comment (jedwards4b) [reply CIME/case/case_clone.py:222]: I think that we have a solution now that fixes the tests and leaves this out - are you okay with that?

Comment (rljacob) [reply CIME/case/case_clone.py:222]: Yes.

**Pull request merged**